### PR TITLE
Convert to V2 Addon

### PR DESCRIPTION
**Pull request by machty**
Notes:

- Minimum version is 3.28 now after getting rid of macros that were used to preserve pre-tracked computed property infrastructure

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,7 +52,8 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        browser: [chrome, firefox]
+        browser: [chrome]
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:55]: Something go wrong with Firefox testing?

File: packages/test-app/ember-cli-build.js [hunk 2]
```diff
@@ -34,19 +34,26 @@ module.exports = function (defaults) {
       // GitHub Pages uses this filename to serve 404s
       emptyFile: '404.html',
     },
+
+    babel: {
+      plugins: [
+        require.resolve(
+          'ember-concurrency/babel-plugin-transform-ember-concurrency-async-tasks.js'
```
Comment (NullVoxPopuli) [inline packages/test-app/ember-cli-build.js:41]: That's a long plugin name!

Thoughts on?:

"ember-concurrency/babel"?

It'd require a a prefix to use without require.resolve, like:

module:ember-concurrency/babel

File: packages/ember-concurrency/babel.config.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Babel config for building the ember-concurrency package itself. Because EC is a V2 addon,
+// this babel config should *not* apply presets or compile away ES modules.
+// It exists only to provide development niceties for you, like automatic
+// template colocation.
+{
+  "plugins": [
+    // https://github.com/embroider-build/embroider/issues/1014
+    ["@babel/plugin-proposal-decorators", { "legacy": true }],
+    "@babel/plugin-proposal-class-properties",
```
Comment (NullVoxPopuli) [inline packages/ember-concurrency/babel.config.json:9]: Any reason not to use the current babel config from the blueprint?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (machty) [reply .github/workflows/ci.yml:55]: Firefox builds are erroring with

> Error: no DISPLAY environment variable specified

Haven't figured out why, will try and remedy before merge.

https://github.com/machty/ember-concurrency/actions/runs/7770826285/job/21191457704

File: packages/test-app/ember-cli-build.js (hunk 2 above)
Comment (machty) [reply packages/test-app/ember-cli-build.js:41]: Ah yeah, I meant to rename this. Will DM you about `module:...`, don't know too much about that.

File: packages/ember-concurrency/babel.config.json (hunk 3 above)
Comment (machty) [reply packages/ember-concurrency/babel.config.json]: I updated to the latest blueprint, thank you!

File: packages/test-app/ember-cli-build.js (hunk 2 above)
Comment (machty) [reply packages/test-app/ember-cli-build.js]: I shortened it; seems to work fine with `require.resolve`:

```
    babel: {
      plugins: [
        require.resolve('ember-concurrency/async-arrow-task-transform'),
      ],
    },
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (machty) [reply .github/workflows/ci.yml]: Restored ff testing; core issue was that I forgot there was a testem.js file that launched FF in headless mode that I forgot to move to test-app folder.

**Pull request merged**